### DISCUSSION TITLE
Add a reference to Schema object inside the response example or response schema

### DISCUSSION THREAD

**Discussion by ArishSultan** [scalar/scalar#1025] [Ideas]
First of all i am really amazed by this tool, it looks so amazing and works like charm, thanks to the team behind this.

I would like to suggest a new feature, i was thinking that it would make sense to add a reference to the actual schema object apart from showing the json schema in the response section.

here is an example from swagger ui.
<img width="594" alt="Screenshot 2024-02-21 at 11 34 22 AM" src="https://github.com/scalar/scalar/assets/31086233/c1fd31b7-e0d2-44aa-a6af-9b31615291f0">

I noticed that scalar actually generate documentation for schema object but they are not reference anywhere at the docs.

looked at this PR https://github.com/scalar/scalar/pull/783, it states that we are following Github, but Github actually does not generate documentation for schema object so, it makes sense to just show the json schema.

**Comment by marclave** [comment]
hey @ArishSultan we appreciate the kind words ✨ 

we have a toggle to show the schema in the response (see screenshot below and also in the sandbox, it's a checkbox in the response container in the right column)

https://sandbox.scalar.com/e/VFlHS
<img width="1512" alt="image" src="https://github.com/scalar/scalar/assets/6176314/d465d360-9b1e-4ea9-9dcd-f6db93c5613c">


was it somewhere else inside of the docs that you wanted the schema to be shown? :)

**Reply by ArishSultan** [reply]
Hi @marclave, thanks for replying fast.

You are right i get the schema through the checkbox but it actually shows the json schema what i am proposing is that it should add a hyperlink or navigation to take me to the actual model that is listed in the models section as in the screenshot.

<img width="1560" alt="Screenshot 2024-02-21 at 2 54 38 PM" src="https://github.com/scalar/scalar/assets/31086233/782eed70-768d-44f9-a8e2-f56757b11e17">

**Reply by ArishSultan** [reply]
Here is a route that returns list of `Media` (specifically `[]Media`) and the schema is shown like this

<img width="1177" alt="Screenshot 2024-02-21 at 3 06 51 PM" src="https://github.com/scalar/scalar/assets/31086233/919bf184-25fc-4dbb-afcb-886afeb72aba">

maybe if we can have it linked with `Media` so it would be easier to track reusable models

**Reply by hanspagel** [reply]
We’re working towards it, but we’ll need to swap out the whole OpenAPI parsing in order to add this feature. 🤓
